test/evp_test.c

```diff
@@ -2092,9 +2092,14 @@ static int kdf_test_init(EVP_TEST *t, const char *name)
 
     kdf = EVP_KDF_fetch(NULL, name, NULL);
     if (kdf == NULL) {
-        OPENSSL_free(kdata);
-        return 0;
+        /*
+         * Don't fail immediately. This algorithm might not be available in this
```

Comment:
I have a PR coming that does 2 things related to this:
(1) make the 'availablein' the first entry (before the command). This then means we can completely skip the init method.
(2) create some is_XX_disabled() methods that can be called before the fetches instead of making the fetch fail everywhere..

Reply:
@slontis - am I best to remove the test aspects of this PR then - and leave it all to your PR? It sounds like yours is broader than mine.

Reply:
Yes that would be good :)